src/compiler/crystal/syntax/lexer.cr

```diff
@@ -2438,6 +2438,11 @@ module Crystal
         next_char
       end
 
+      if start == current_pos
+        @token.type == :EOF
```

Comment:
is this meant to be a comparison, not an assignment?

Reply:
Oops, fixed!